### ISSUE TITLE
inconsistent results with getSession()

### ISSUE THREAD

**Issue by electronicalias** [amazon-connect/amazon-connect-streams#776]
**Summary**: getSession() doesn't return consistent results despite the apparent consistency in calls.

**Context**: We are creating an interface that allows us to transcribe the remote audio on the page using a 3rd party transcribing service. Therefore we need to access the `_remoteAudioStream` to send it to the transcriber.. 

**Expected Behaviour**: `_remoteAudioStream` is consistently returned as part of the rtcSession object.

**Actual Behaviour**: `_remoteAudioStream` can be missing or `getSession()` returns `undefined` despite the fact that a call is active and working. Once this happens, without clearing contacts/refresh/login again we cannot get a full session response, rendering our application valueless.

**Examples**:
An example of a successful call to `getSession()` with a call connected using the following code:
```javascript
export function getMediaStream() {
    var connectionId = window.myCCP.contact.getAgentConnection()['connectionId']
    
    if (connectionId) {
        console.log("Got connectionId: " + connectionId)
        var softphoneManager = connect.core.getSoftphoneManager();
        if (softphoneManager) {
            console.log("sofphoneManager exists")
            for (const key in softphoneManager) {
                if (softphoneManager.hasOwnProperty(key)) {
                const value = softphoneManager[key];
                console.log(`Key: ${key}, Value: ${value}`);
                }
            }
            var session = softphoneManager.getSession(connectionId)
            if(session) {
                console.log("session exists!")
                for (const key in session) {
                    if (session.hasOwnProperty(key)) {
                    const value = session[key];
                    console.log(`Key: ${key}, Value: ${value}`);
                    }
                }
            }
            
            localStorage.setItem('session', session);
        }
        
    }
}
```

NOTE: any console.log(x) is purely to show what's happening.

![image](https://github.com/amazon-connect/amazon-connect-streams/assets/3788860/e3a84d08-7e32-49db-acfe-b4f5417399c6)

When it is not working, with a call connected, we can run this function (attached to a button) and it returns `undefined`. 

`connectionId` and `softphoneManager` are returned, but `getSession` is not.

**Comment by electronicalias** [comment]
I've done some further investigation into this. When the error condition is present, I'm also getting a signalling_handshake_failure:
`"Softphone error occurred :  signalling_handshake_failure Handshaking with Signalling Server wss://rtc.cell-1.prod.eu-west-2.prod.connect.aws.a2z.com/LilyRTC failed. "`

This is intermittent. It is only when I get this that `getSession` fails to return the data I need. However the call is still connected and working perfectly. It is not a firewall issue, I can use the standard inbuilt CCP without ever seeing any issues to do with signalling.

On a failing session, the following code is run twice:
https://github.com/amazon-connect/amazon-connect-streams/blob/bd0b301b1eb97d0e5f72db1937d7ad8fa19d2a80/src/softphone.js#L83

I setup a condition in `contact.onConnected`, when I get a working session the information is returned, when I do not, there is no connection information - however as stated the call is still connected and audio is working both ways - so where is the `RTCPeerConnection` being stored in this instance:

```javascript
var sd = globalThis.myCCP.session._pc.remoteDescription
if (sd) {
    console.log(`handshake: Remote session: type='${sd.type}'; sdp description='${sd.sdp}'`);
} else {
    console.log("handshake: No remote session yet.");
}
```

I have also noticed that, using the current version, vs. a much older version `2.1.1` if I remember correctly (forgotten how to find out where the version of the release is), it is more stable, but still not stable enough for the application we intend to use it for.

It doesn't matter if we're using a locally hosted CCP, a CCP hosted in S3, or the CCP hosted in S3 and driven by Selenium in a container, we get the same intermittent results.

**Comment by electronicalias** [comment]
We did some further analysis and found that we were getting the following message (through debugging) that returns the signalling error:
'DUPLICATE_RTC_INVITE_REQUEST'

When we get this, despite audio being connected, the presence of _remoteAudioStream is missing, ergo our software then cannot transcribe. 

We implemented a test for _remoteAudioStream and if not available, with exponential backoff, we timeout, hangup, redial and recheck - usually within 1-2 retries, we are able to get _remoteAudioStream. 

This is however a work around and we can't understand/find the code that actually creates the RTCPeerConnection - logically the _remoteAudioStream has to exist even in an error state given the fact we can hear remote end of the conversation, however as stated it is not in the expected place.

The second work around is to create an object in the browser, by adding lines to the connect-streams.js. Specifically after the block below:

if (rtcPeerConnectionFactory) {
        session.connect(rtcPeerConnectionFactory.get(callConfig.iceServers));
        var rtcConnDetail = rtcPeerConnectionFactory.get(callConfig.iceServers);
        
      } else {
        session.connect();
      };

      // trying to add it here instead.. 
      globalThis.myCCP.session = session;

What we do not understand is where the rtcPeerConnectionFactory stores the RTCPeerConnection object, therefore we have to copy it to globalThis.myCCP.session.

During the call this is updated as the rtcPeerConnectionFactory handler is called - what we really need to know to avoid having to modify connect-streams.js is where the original/source RTCPeerConnection object is stored so we can call against it directly.
